UnitsNet.Tests/GeneratedCode/TestsBase/AccelerationTestsBase.g.cs

```diff
@@ -292,20 +294,113 @@ public void As()
         }
 
         [Fact]
-        public void As_SIUnitSystem_ThrowsArgumentExceptionIfNotSupported()
+        public virtual void BaseUnit_HasSIBase()
+        {
+            var baseUnitInfo = Acceleration.Info.BaseUnitInfo;
+            Assert.True(baseUnitInfo.BaseUnits.IsSubsetOf(UnitSystem.SI.BaseUnits));
+        }
+
+        [Fact]
+        public virtual void As_UnitSystem_SI_ReturnsQuantityInSIUnits()
+        {
+            var quantity = new Acceleration(value: 1, unit: Acceleration.BaseUnit);
+            var expectedValue = quantity.As(Acceleration.Info.GetDefaultUnit(UnitSystem.SI));
+
+            var convertedValue = quantity.As(UnitSystem.SI);
+
+            Assert.Equal(expectedValue, convertedValue);
+        }
+
+        [Fact]
+        public void As_UnitSystem_ThrowsArgumentNullExceptionIfNull()
+        {
+            var quantity = new Acceleration(value: 1, unit: Acceleration.BaseUnit);
+            UnitSystem nullUnitSystem = null!;
+            Assert.Throws<ArgumentNullException>(() => quantity.As(nullUnitSystem));
+        }
+
+        [Fact]
+        public void As_UnitSystem_ThrowsArgumentExceptionIfNotSupported()
+        {
+            var quantity = new Acceleration(value: 1, unit: Acceleration.BaseUnit);
+            // simulating a combination of units that won't match anything
+            var unsupportedUnitSystem = new UnitSystem(new BaseUnits(
+                (LengthUnit)(-1), (MassUnit)(-1), (DurationUnit)(-1), (ElectricCurrentUnit)(-1), (TemperatureUnit)(-1), (AmountOfSubstanceUnit)(-1), (LuminousIntensityUnit)(-1)));
+            Assert.Throws<ArgumentException>(() => quantity.As(unsupportedUnitSystem));
```

Comment:
It seems this particular test-only BaseUnits of all values `-1` is used alot for testing (yes yes, I realize we are generating tests), maybe we can move it to a static readonly named something like `TestConstants.UnsupportedBaseUnit` and reuse it?

Mostly for readability, and we can move comment to xmldoc of static field.

Reply:
Fixed in https://github.com/angularsen/UnitsNet/pull/1473/commits/4fc87b47ed74847fbb29eca5a936dab93107b4cc (I put it in the `QuantityTestsBase`, if you want a `TestConstants` class- pick the location)

Reply:
The locatino is fine.

Also, this is nitpicking, but I realize we could just also provide `new UnitSystem(UnsupportedBaseUnits)` in a similar `UnsupportedUnitSystem` singleton instance, since both UnitSystem and BaseUnits are basically immutable.

Reply:
Yeah, I know- but I wanted to have the type visible (otherwise it's not obvious what type we're giving as input).